#version: '3.8'

services:
  # Odoo Service
  odoo:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: odoo_app
    ports:
      - "8069:8069"
      - "8072:8072"
    volumes:
      - ./odoo_data:/var/lib/odoo
      - ./custom_modules:/mnt/custom_modules
      - ./logs:/var/log/odoo
      - ./geoip:/usr/share/geoip
      - ./odoo.conf.template:/etc/odoo/odoo.conf.template
    environment:
      - DB_HOST=${DB_HOST}
      - DB_PORT=${DB_PORT}
      - DB_USER=${DB_USER}
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_NAME=${DB_NAME}
      - ADMIN_PASSWORD=${ADMIN_PASSWORD}
      - LC_ALL=C.UTF-8
      - LANG=C.UTF-8
    restart: unless-stopped # Thay đổi từ always để tránh restart liên tục khi có lỗi
    networks:
      - odoo_network
    user: odoo:odoo
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8069/web/database/selector" ]
      interval: 30s
      timeout: 10s
      retries: 3
      start_period: 40s

volumes:
  odoo_data:
    driver: local

networks:
  odoo_network:
    driver: bridge
